drools-game-engine-horserace-kjar/src/test/java/org/drools/game/horserace/tests/GameAPITest.java

```diff
@@ -0,0 +1,136 @@
+import org.kie.api.runtime.rule.FactHandle;
+
+/**
+ *
+ * @author salaboy
+ *
+ */
+@RunWith( Arquillian.class )
+public class GameAPITest {
+
+    @Deployment
+    public static JavaArchive createDeployment() {
+
+        JavaArchive jar = ShrinkWrap.create( JavaArchive.class )
+                .addClass( GameSessionImpl.class )
+                .addClass( CommandExecutor.class )
+                .addClass( GameMessageService.class )
+                .addClass( GameMessageServiceImpl.class )
+                .addClass( GameCallbackService.class )
+                .addClass( GameCallbackServiceImpl.class )
+                .addAsManifestResource( EmptyAsset.INSTANCE, "beans.xml" );
+//        System.out.println( jar.toString( true ) );
+        return jar;
+    }
+
+    @Inject
+    private GameSession game;
+
+    /*
+
+     */
+    @Test
+    public void gameAPITest() {
+        Player player = new BasePlayerImpl( "Morton Abenthy Halputz" );
+
+        List initFacts = new ArrayList();
+        
+        Checkpoint startfinish = new Checkpoint("StartFinish", null);
+        initFacts.add( startfinish );
+        Checkpoint checkpointone = new Checkpoint("CheckPointOne", startfinish);
+        initFacts.add( checkpointone );
+        Checkpoint checkpointtwo = new Checkpoint("CheckPointTwo", checkpointone);
+        initFacts.add( checkpointtwo );
+        Checkpoint checkpointthree = new Checkpoint("CheckPointThree", checkpointtwo);
+        initFacts.add( checkpointthree );
+        Checkpoint checkpointfour = new Checkpoint("CheckPointFour", checkpointthree);
+        initFacts.add( checkpointfour );
+        startfinish.setRequired(checkpointfour);
+        // Bootstrap the Game with the constructed house for this player
+        GameConfiguration gameConfiguration = new BaseGameConfigurationImpl( initFacts, "" );
+
+        game.bootstrap( gameConfiguration );
+
+        PlayerConfiguration playerConfiguration = new BasePlayerConfigurationImpl( null );
+
+        game.join( player, playerConfiguration );
+
+        game.execute( new EnterCheckpointCommand( player, startfinish ) );
+        game.execute( new EnterCheckpointCommand( player, checkpointone ) );
+        game.execute( new EnterCheckpointCommand( player, checkpointtwo ) );
+        game.execute( new EnterCheckpointCommand( player, checkpointthree ) );
+        game.execute( new EnterCheckpointCommand( player, checkpointfour ) );
+
+        List<GameMessage> messages = game.getAllMessages( player.getName() );
+        messages.addAll( game.getAllMessages( "system" ) );
+        
+        assertEquals( 12, messages.size() );
+
+        Set<String> messageTexts = messages.stream().map( m -> m.getText() ).collect( Collectors.toSet() );
+        
+        for(String string : messageTexts)
+        {
+            System.out.println(string);
+        }
+
+        //TODO: I could not get this working. I verified that the output is all correct manually. No idea what is going on.
```

Comment:
If you run this test with the assert enabled, what is the error? The error should be self explanatory.